.gitmodules

```diff
@@ -0,0 +1,9 @@
+[submodule "tests/gif"]
```

Comment:
Not a blocker, but a development opinion of mine: I like to have the CI and/or testing script(s) pull `git` repositories for themselves, rather than using explicit submodules. Two reasons for this:

* `git` novices are often confused by submodules and get frustrated/give up when they can't get tests to work locally (usually because they forgot to clone `--with-submodules` or whatever the flag is)
* There's still no standard-ish way to remove submodules once registered -- they stick around as bits of metadata and extra state in `.git/` even after they're removed from `.gitmodules`. 

That being said, they're a lot easier to set up (and faster) than cloning a bunch of repos in a shell script for the CI. So no need to change, just thought I'd put my 0.02c in.

Reply:
These are good points.

At some point, I want to rewrite the test runner in Rust instead of Bash. I will try to keep these points in mind for then. For the interim, I think submodules are a reasonable solution. So I am going to merge.